tasks/install_dist_pkgs.yml

```diff
@@ -74,27 +81,71 @@
     runrefresh: "yes"
     auto_import_keys: "yes"
     disable_gpg_check: "yes"
-  when: nrinfragent_os_name == 'suse'
+  when: nrinfragent_os_name|lower == 'suse'
   register: infra_agent_zypper_repo
-
+  
 - name: run make cache to actually import gpg key
   command: "yum -q makecache -y --disablerepo='*' --enablerepo='newrelic-infra'"
-  when: nrinfragent_os_name == 'redhat' and setup_agent_repo.changed
+  when: nrinfragent_os_name|lower == 'redhat' and (setup_agent_repo_amazon.changed or setup_agent_repo_redhat.changed)
   tags:
     - skip_ansible_lint
   args:
     warn: false
 
-- name: install agent
+- name: create log destination directory
+  file:
+    path: "{{ nrinfragent_config['log_file'] | dirname }}"
+    state: directory
+    recurse: true
+  when: 
+    - nrinfragent_state != 'absent'
+    - nrinfragent_config['log_file'] is defined
+
+- name: install agent pre2.8.0
+  package:
+    name: "newrelic-infra"
+    state: "{{ nrinfragent_state }}"
+  when: nrinfragent_os_name|lower != 'windows' and ansible_version.full < "2.8.0"
+
+- name: install agent post2.8.0
   package:
     name: "newrelic-infra"
     state: "{{ nrinfragent_state }}"
-  when: nrinfragent_os_name != 'windows'
+  when: nrinfragent_os_name|lower != 'windows' and ansible_version.full >= "2.8.0" and nrinfragent_os_name|lower != 'amazon' and nrinfragent_os_name|lower != 'redhat'
```

Comment:
@JakubKotkowiak wasn't the purpose of breaking out the pre/post in order to use the lock timeout?  If so, I don't see the lock timeout specified here.

Reply:
Yes, but in this particular case it's for systems that don't support lock_timeout